hathor/dag_builder/types.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from typing import Iterator, NamedTuple
+
+from hathor.transaction import BaseTransaction
+from hathor.wallet import BaseWallet
+
+AttributeType = dict[str, str | int]
+VertexResolverType = Callable[[BaseTransaction], None]
+WalletFactoryType = Callable[[], BaseWallet]
```

Comment:
I recommend typing those with `TypeAlias`, to help mypy.

Reply:
Done!